src/client.rs

```diff
@@ -578,12 +558,13 @@ where
             //     Persist timestamp metadata. The client MUST write the file to non-volatile
             //     storage as FILENAME.EXT (e.g. timestamp.json).
 
-            self.store_metadata(
-                &timestamp_path,
-                &MetadataVersion::None,
-                &raw_signed_timestamp,
-            )
-            .await;
+            self.local
```

Comment:
There are similar issues with the ordering of trusting/persisting timestamp, snapshot, and targets metadata.  Though, in these instances, the spec is not clear as to an ordering of trusting/persisting these metadata.

As-is has an interesting property that if a client's local repository is completely broken and always fails stores, pounding on update() will eventually fetch and trust the most recent targets metadata, allowing clients to eventually resolve targets (once update() returns `Ok(false)` instead of `Err(_)`).  Is that preferable to not updating the trusted metadata when storing the metadata fails (so update() continuously returns `Err(_)` and target resolution never unblocks)?